Delete dead code IOUtil.compress and decompress [HZ-3350]

orcunc opened the pull request:
Jira : https://hazelcast.atlassian.net/browse/HZ-3550

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [x] Request reviewers if possible

vbekiaris reviewed (changes requested):
We are using `compress` & `decompress` methods in the new UCD namespaces feature (and deduplicate some `DeflaterOutputStream` usages in `JobRepository` by using `compress` instead).

orcunc commented:
> We are using `compress` & `decompress` methods in the new UCD namespaces feature (and deduplicate some `DeflaterOutputStream` usages in `JobRepository` by using `compress` instead).

So it is not dead code anymore. Right ?

orcunc commented:
> > We are using `compress` & `decompress` methods in the new UCD namespaces feature (and deduplicate some `DeflaterOutputStream` usages in `JobRepository` by using `compress` instead).
> 
> So it is not dead code anymore. Right ?

@vbekiaris Do you think we should change the code in compress to be as below instead of assuming compressed length is 1 /10 th of input. Because if the assumption does not hold, there will be a memory resizing

`ByteArrayOutputStream bos = new ByteArrayOutputStream(input.length);`

vbekiaris commented:
@orcunc good point, I ran a few tests for our typical use case of `compress`, which is to compress class file bytes.

With `DEFAULT_COMPRESSION` level, average compressed/original ratio is ~47%, with 90% of compressed files being < 56% the original size. So I think we can change the default size to 60% of the original and we should be mostly resize-free (60% is around the 94th percentile in my samples).

orcunc commented:
> @orcunc good point, I ran a few tests for our typical use case of `compress`, which is to compress class file bytes.
> 
> With `DEFAULT_COMPRESSION` level, average compressed/original ratio is ~47%, with 90% of compressed files being < 56% the original size. So I think we can change the default size to 60% of the original and we should be mostly resize-free (60% is around the 94th percentile in my samples).

Something like this ? 
1. I have changed the default buffer size to be %60 of the original. 
2. Changed compression level to be DEFAULT_COMPRESSION. 
3. Or do you want to keep the compression level as is.  Deflater.BEST_SPEED. 

```java
public static byte[] compress(byte[] input) {
        if (input.length == 0) {
            return new byte[0];
        }
        // 60% of the original length
        int len = (int)(input.length * 0.6);
        len = Math.max(len, 10); 
        
        Deflater compressor = new Deflater();
        compressor.setLevel(Deflater.DEFAULT_COMPRESSION);  // <-- Changed
        compressor.setInput(input);
        compressor.finish();
        ByteArrayOutputStream bos = new ByteArrayOutputStream(len);
        byte[] buf = new byte[len];
        while (!compressor.finished()) {
            int count = compressor.deflate(buf);
            bos.write(buf, 0, count);
        }
        compressor.end();
        return bos.toByteArray();
    }
```

vbekiaris commented:
> > @orcunc good point, I ran a few tests for our typical use case of `compress`, which is to compress class file bytes.
> > With `DEFAULT_COMPRESSION` level, average compressed/original ratio is ~47%, with 90% of compressed files being < 56% the original size. So I think we can change the default size to 60% of the original and we should be mostly resize-free (60% is around the 94th percentile in my samples).
> 
> Something like this ?
> 
> 1. I have changed the default buffer size to be %60 of the original.
> 2. Changed compression level to be DEFAULT_COMPRESSION.
> 3. Or do you want to keep the compression level as is.  Deflater.BEST_SPEED.
> 
> ```java
> public static byte[] compress(byte[] input) {
>         if (input.length == 0) {
>             return new byte[0];
>         }
>         // 60% of the original length
>         int len = (int)(input.length * 0.6);
>         len = Math.max(len, 10); 
>         
>         Deflater compressor = new Deflater();
>         compressor.setLevel(Deflater.DEFAULT_COMPRESSION);  // <-- Changed
>         compressor.setInput(input);
>         compressor.finish();
>         ByteArrayOutputStream bos = new ByteArrayOutputStream(len);
>         byte[] buf = new byte[len];
>         while (!compressor.finished()) {
>             int count = compressor.deflate(buf);
>             bos.write(buf, 0, count);
>         }
>         compressor.end();
>         return bos.toByteArray();
>     }
> ```

Yes, 60% and `DEFAULT_COMPRESSION` level will be fine, thank you Orcun!

orcunc commented:
> > > @orcunc good point, I ran a few tests for our typical use case of `compress`, which is to compress class file bytes.
> > > With `DEFAULT_COMPRESSION` level, average compressed/original ratio is ~47%, with 90% of compressed files being < 56% the original size. So I think we can change the default size to 60% of the original and we should be mostly resize-free (60% is around the 94th percentile in my samples).
> > 
> > 
> > Something like this ?
> > 
> > 1. I have changed the default buffer size to be %60 of the original.
> > 2. Changed compression level to be DEFAULT_COMPRESSION.
> > 3. Or do you want to keep the compression level as is.  Deflater.BEST_SPEED.
> > 
> > ```java
> > public static byte[] compress(byte[] input) {
> >         if (input.length == 0) {
> >             return new byte[0];
> >         }
> >         // 60% of the original length
> >         int len = (int)(input.length * 0.6);
> >         len = Math.max(len, 10); 
> >         
> >         Deflater compressor = new Deflater();
> >         compressor.setLevel(Deflater.DEFAULT_COMPRESSION);  // <-- Changed
> >         compressor.setInput(input);
> >         compressor.finish();
> >         ByteArrayOutputStream bos = new ByteArrayOutputStream(len);
> >         byte[] buf = new byte[len];
> >         while (!compressor.finished()) {
> >             int count = compressor.deflate(buf);
> >             bos.write(buf, 0, count);
> >         }
> >         compressor.end();
> >         return bos.toByteArray();
> >     }
> > ```
> 
> Yes, 60% and `DEFAULT_COMPRESSION` level will be fine, thank you Orcun!

@vbekiaris , @JackPGreen and @arodionov please find the new code. I have changed the compress() code and fixed some warnings in tests

orcunc commented:
Thank everybody
